### DISCUSSION TITLE
How to create a custom matcher for snapshot testing?

### DISCUSSION THREAD

**Discussion by gsipos** [vitest-dev/vitest#2818] [Q&A] [answered]
I have a React component, let's call it `MyComponent`. If I want to write a snapshot test for it, then it woul look like this:

```typescript
it('matches snapshot', () => 
  expect(render(
<MyAppThemeProvider>
  <MyComponent />
</MyAppThemeProvider>
).asFragment()).toMatchSnapshot());
```

I would like to hide the boilerplate of the testing-libraries `render(...).asFragment()` and having to wrap everything with `<MyAppThemeProvider>`.

My question is, is it possible to write a custom matcher so my test would look like this:

```typescript
it('matches snapshot', () => expect(<MyComponent />).toMatchSnapshotWithMyAppTheme());
```

If not, then what is the best practice you suggest to hide this kind of boilerplate?

**Comment by sheremet-va** [comment, marked as answer]
Vitest doesn't support custom snapshot matchers. But you can add your own serializer with `expect.addSnapshotSerializer`: https://vitest.dev/guide/snapshot.html#custom-serializer

**Reply by gsipos** [reply]
Thanks for the quick answer!
